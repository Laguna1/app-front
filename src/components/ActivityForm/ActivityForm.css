.a-form {
  color: crimson;
  padding: 20px;
}
