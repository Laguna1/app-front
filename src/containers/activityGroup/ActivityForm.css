.main {
  background-color: cornflowerblue;
  color: crimson;
  border: 2px solid blue;
}
