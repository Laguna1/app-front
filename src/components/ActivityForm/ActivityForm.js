import React from 'react';
import PropTypes from 'prop-types';
import './ActivityForm.css';
import { connect } from 'react-redux';
import { updateActivity } from '../../actions/activity';

class ActivityForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      place: '',
      intensity: '',
    };
  }

  componentDidMount = () => {
    const { actionToPerform, buttonId, activities } = this.props;
    if (actionToPerform === 'Save Changes') {
      const activity = activities.filter((x) => x.id.toString() === buttonId);
      this.setState({
        name: activity[0].name,
        place: activity[0].place,
        intensity: activity[0].intensity,
      });
    }
  }

  handleChangeName = (e) => {
    this.setState({
      name: e.target.value,
    });
  }

  handleChangePlace = (e) => {
    this.setState({
      place: e.target.value,
    });
  }

  handleChangeIntensity = (e) => {
    this.setState({
      intensity: e.target.value,
    });
  }

  handleSubmit = (name, place, intensity) => {
    const { addActivity } = this.props;
    addActivity(name, place, intensity);
  }

  handleUpdate = async (id) => {
    const { name, place, intensity } = this.state;

    const {
      user, updateActivity, changeEditForm,
    } = this.props;

    const data = {
      id,
      userId: user.user.id,
      name,
      place,
      intensity,
    };

    await updateActivity(data);
    changeEditForm();
  }

  render() {
    const { name, place, intensity } = this.state;
    const {
      actionToPerform, activities, buttonId, changeEditForm, changeAddForm,
    } = this.props;
    const activity = activities.filter((x) => x.id.toString() === buttonId);
    return (
      <div>
        <h3>
          {actionToPerform}
          {' '}
          Activity
        </h3>

        <form
          className="one-form"
          onSubmit={
           actionToPerform === 'Add'
             // eslint-disable-next-line max-len
             ? () => this.handleSubmit(name, place, intensity) : () => this.handleUpdate(activity[0].id)
}
        >
          <div className="one-parameter">
            <label htmlFor="name">
              Name:
              <input
                required
                id="name"
                type="text"
                name="name"
                defaultValue={buttonId === '0' ? name : activity[0].name}
                onChange={this.handleChangeName}
              />
            </label>
          </div>
          <div className="one-parameter">
            <label htmlFor="place">
              Place:
              <input
                id="place"
                name="place"
                defaultValue={buttonId === '0' ? place : activity[0].place}
                onChange={this.handleChangePlace}
              />
            </label>

          </div>
          <div className="one-parameter">
            <label htmlFor="intensity">
              Intensity:
              <input
                id="intensity"
                name="intensity"
                defaultValue={buttonId === '0' ? intensity : activity[0].intensity}
                onChange={this.handleChangeIntensity}
              />
            </label>

          </div>
          <div className="buttons-form">
            {actionToPerform === 'Add' && <button type="submit">{actionToPerform}</button>}
            {actionToPerform === 'Save Changes' && <button type="submit">Save</button>}
            {actionToPerform === 'Add' && <button type="button" onClick={changeAddForm}>Cancel</button>}
            {actionToPerform === 'Save Changes' && <button type="button" onClick={changeEditForm}>Cancel</button>}
          </div>
        </form>
      </div>
    );
  }
}

ActivityForm.propTypes = {
  addActivity: PropTypes.func,
  actionToPerform: PropTypes.string,
  changeAddForm: PropTypes.func,
  activities: PropTypes.instanceOf(Array),
  buttonId: PropTypes.string,
  updateActivity: PropTypes.func,
  changeEditForm: PropTypes.func,
  user: PropTypes.shape({
    user: PropTypes.shape({
      id: PropTypes.number,
    }),
  }),

};

ActivityForm.defaultProps = {
  changeAddForm: () => {},
  addActivity: () => {},
  actionToPerform: '',
  activities: [],
  buttonId: '0',
  updateActivity: () => {},
  changeEditForm: () => {},
  user: {},
};

const mapStateToProps = (state) => ({
  user: state.user,
  activities: state.activity,
});
const mapDispatchToProps = (dispatch) => ({
  updateActivity: (data) => dispatch(updateActivity(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ActivityForm);
