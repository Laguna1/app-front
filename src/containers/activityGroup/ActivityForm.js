import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ActivityForm from '../../components/ActivityForm/ActivityForm';
import {
  fetchUserActivity, createActivity, deleteActivity,
} from '../../actions/activity';
import { loginStatus } from '../../actions/user';
import './ActivityForm.css';

class Activity extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      addForm: false,
      editForm: false,
      idActivity: '0',
    };
  }

  componentDidMount() {
    const { user, fetchUserActivity } = this.props;
    const ID = user.user.id;
    fetchUserActivity(ID);
  }

  shouldComponentUpdate(nextProps, nextState) {
    const { activity } = this.props;
    const {
      addForm, editForm,
    } = this.state;
    return activity !== nextProps.activity
    || addForm !== nextState.addForm
    || editForm !== nextState.editForm;
  }

  addActivity = (name, description) => {
    const { createActivity, user } = this.props;
    const { addForm } = this.state;
    const userId = user.user.id;
    createActivity({ name, description, userId });
    this.setState({
      addForm: !addForm,
    });
  };

  displayForm = () => {
    const { addForm } = this.state;
    this.setState({
      addForm: !addForm,
    });
  }

  displayEdit= (e) => {
    const { editForm } = this.state;
    this.setState({
      editForm: !editForm,
      idActivity: e.target.id,
    });
  }

  deleteActivity = (id) => {
    const { user } = this.props;
    const { deleteActivity } = this.props;
    const userId = user.user.id;
    deleteActivity({ userId, id });
  }

   changeEditForm = () => {
     const { editForm } = this.state;
     this.setState({
       editForm: !editForm,
     });
   }

   changeAddForm = () => {
     const { addForm } = this.state;
     this.setState({
       addForm: !addForm,
     });
   }

   render() {
     const { activity } = this.props;
     const {
       addForm, editForm, idActivity,
     } = this.state;
     return (
       <main className="main">
         <button type="button" className="add-activity" onClick={this.displayForm}>+</button>
         <div className="activities">
           { !editForm && !addForm && <h3>Your Activities</h3>}

           {activity.length === 0 && !addForm && <div className="tracking">Start adding a illness you want to track here!</div>}
           {activity.map((activ) => (
             <div key={activ.id}>
               { !editForm && !addForm && (
               <div className="one-activity">
                 <div className="buttons">
                   <button type="button" onClick={this.displayEdit}>
                     <i className="fa fa-pencil-square-o" id={activity.id} />
                   </button>
                   <button type="button" onClick={() => this.deleteActivity(activity.id)}>
                     <i className="fa fa-trash-o" />
                   </button>

                 </div>
                 <div className="activity-info">
                   <Link to={{
                     pathname: `activity/${activity.id}`,
                     state: {
                       nameactivity: activity.name,
                     },
                   }}
                   >
                     {!editForm && (
                     <div>
                       <div className="activity-name">
                         <p>Name:</p>
                         <p>Place:</p>
                       </div>
                       <div className="activity-place">
                         <p>{activity.name}</p>
                         <p>{activity.place}</p>
                       </div>
                     </div>
                     )}
                   </Link>
                 </div>
               </div>
               )}
               { editForm && activity.id.toString() === idActivity && <ActivityForm actionToPerform="Save Changes" buttonId={idActivity} changeEditForm={this.changeEditForm} />}
             </div>
           ))}
         </div>
         <div className="new-activity">
           {addForm && <ActivityForm addActivity={this.addActivity} actionToPerform="Add" changeAddForm={this.changeAddForm} />}
         </div>
       </main>
     );
   }
}

const mapStateToProps = (state) => (
  {
    user: state.user,
    isLogin: state.user.isLogin,
    activity: state.activity,
  });
const mapDispatchToProps = (dispatch) => ({
  fetchUserActivity: (data) => dispatch(fetchUserActivity(data)),
  createActivity: (data) => dispatch(createActivity(data)),
  deleteActivity: (id) => dispatch(deleteActivity(id)),
  loginStatus: () => dispatch(loginStatus()),
});

Activity.propTypes = {
  fetchUserActivity: PropTypes.func,
  createActivity: PropTypes.func,
  deleteActivity: PropTypes.func,
  user: PropTypes.shape({
    user: PropTypes.shape({
      id: PropTypes.number,
    }),
  }),
  activity: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    place: PropTypes.string,
  })),
};

Activity.defaultProps = {
  createActivity: () => {},
  deleteActivity: () => {},
  fetchUserActivity: () => {},
  activity: {},
  user: {},
};

export default connect(mapStateToProps, mapDispatchToProps)(Activity);
