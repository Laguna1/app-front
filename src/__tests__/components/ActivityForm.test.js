import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import combinedReducers from '../../reducers';
import ActivityForm from '../../components/ActivityForm/ActivityForm';

test('ActivityForm renders correctly', () => {
  const store = createStore(combinedReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <ActivityForm />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
