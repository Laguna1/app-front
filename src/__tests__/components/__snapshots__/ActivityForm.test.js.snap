// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityForm renders correctly 1`] = `
<div>
  <h3>
    
     
    Activity
  </h3>
  <form
    className="one-form"
    onSubmit={[Function]}
  >
    <div
      className="one-parameter"
    >
      <label
        htmlFor="name"
      >
        Name:
        <input
          defaultValue=""
          id="name"
          name="name"
          onChange={[Function]}
          required={true}
          type="text"
        />
      </label>
    </div>
    <div
      className="one-parameter"
    >
      <label
        htmlFor="place"
      >
        Place:
        <input
          defaultValue=""
          id="place"
          name="place"
          onChange={[Function]}
        />
      </label>
    </div>
    <div
      className="one-parameter"
    >
      <label
        htmlFor="intensity"
      >
        Intensity:
        <input
          defaultValue=""
          id="intensity"
          name="intensity"
          onChange={[Function]}
        />
      </label>
    </div>
    <div
      className="buttons-form"
    />
  </form>
</div>
`;
